src/main/java/com/commercetools/project/sync/category/service/CategoryReferenceTransformService.java

```diff
@@ -0,0 +1,14 @@
+package com.commercetools.project.sync.category.service;
+
+import io.sphere.sdk.categories.Category;
+import io.sphere.sdk.categories.CategoryDraft;
+import java.util.List;
+import java.util.concurrent.CompletionStage;
+import javax.annotation.Nonnull;
+
+public interface CategoryReferenceTransformService {
+
+  @Nonnull
+  CompletionStage<List<CategoryDraft>> transformCategoryReferences(
```

Comment:
This structure I have implemented before our discussion and I don't think it is useful now to spend extra time on this ticket to correct the structure or design. Because after implementing our new design in Java Sync, We can create a follow up ticket in Project sync to correct this structure and implement according to the Java Sync changes.
Please let me know if this is really important now to change it? Then I might again need some extra time on the ticket.

Reply:
I see, for now, let's keep it as is 👍🏽 
If it's not a problem for you, you might put todos for potential changes in code so we will not forget later. 

As said before, I did not review this PR's diff but only commented about structure. I will try to check the diff and coverage today when I have time.

Reply:
Okay Thanks, Yes I will add TODO.